apps/graphql/src/apps/itinerary/dataloaders/Itineraries.js

```diff
@@ -53,17 +53,23 @@ export const parseParameters = (input: ItinerariesSearchParameters) => {
 };
 
 export const parseParametersNew = (
-  input: ItinerariesOneWaySearchParameters, // @TODO: Later on will be expanded to be the return one with extra fields
+  input: ItinerariesReturnSearchParameters,
```

Comment:
This can handle oneway and return, right? So probably more generic name will be better, for better understanding.

Reply:
What do you propose @FilipMessa  ? I think maybe `ItinerariesRoundTripSearchParameters` would be nice. However we discussed with @RobinCsl that moving forward we should make it a union type and then add the specific parameters for each of the search queries in separate functions depending on the input (or Flow complains as for example `inboundDate` does not exist when the input type is OneWay)

Reply:
Can be GraphQL input type as the union type?

Reply:
GraphQL does not support union for input types.

Reply:
The `ItinerariesSearchParameters` will be the best but we already use it for old query so maby just `ItinerariesInputSearchParameters`? 🤔

Reply:
@michalsanger Yes but this is about the flow union types. As I realized after discussion with @aalises

Reply:
Oh sorry, thanks for info :-)